nsot/api/views.py

```diff
@@ -504,23 +519,41 @@ def next_network(self, request, pk=None, site_pk=None, *args, **kwargs):
         prefix_length = params.get('prefix_length')
         num = params.get('num')
         strict = qpbool(params.get('strict_allocation', False))
-        networks = network.get_next_network(
-            prefix_length, num, strict, as_objects=False
-        )
-
-        return self.success(networks)
+        if request.method == 'POST':
+            state = 'reserved' if qpbool(params.get('reserve', False)) else 'allocated'
+            is_ip = True if prefix_length == 32 or prefix_length == 128 else False
+            networks = network.get_next_network(
+                prefix_length, num, strict, as_objects=True
+            )
+            self.allocate_networks(networks, prefix_length, site_pk, network, state=state, is_ip=is_ip)
+        else:
+            networks = network.get_next_network(
```

Comment:
Suggestion: Since you're returning the `unicode` representation of the Networks in all cases, you could simplify this code by getting rid of this `else` and then set `networks` with `as_objects=True` right before the if statement.